packages/astro/src/runtime/server/index.ts

```diff
@@ -480,11 +480,11 @@ export async function renderHead(result: SSRResult): Promise<string> {
 			}
 			return renderElement('script', {
 				...script,
-				props: { ...script.props, 'astro-script': result._metadata.pathname + '/script-' + i },
+				props: { ...script.props, 'astro-script': result._metadata.legacyBuild ? result._metadata.pathname + '/script-' + i : undefined },
```

Comment:
This causes deduping issues for props that are otherwise the same. Won't be needed once `legacyBuild` is dropped.